Detectors/GlobalTracking/src/MatchITSTPCQC.cxx

```diff
@@ -158,6 +197,23 @@ bool MatchITSTPCQC::init()
     mEtaVsPtNum[i] = new TH2F(Form("mEtaVsPtNum_%s", title[i].c_str()), Form("Eta vs Pt distribution of ITSTPC matched tracks, wrt %s tracks; #it{p}_{T} [GeV#it{c}]; #it{p}_{T} [GeV#it{c}]; Eta", title[i].c_str()), 100, 0.f, 20.f, 100, -2.f, 2.f);
     mEtaVsPtDen[i] = new TH2F(Form("mEtaVsPtDen_%s", title[i].c_str()), Form("Eta vs Pt distribution of %s tracks; #it{p}_{T} [GeV#it{c}]; #it{p}_{T} [GeV#it{c}]; Eta", title[i].c_str()), 100, 0.f, 20.f, 100, -2.f, 2.f);
     mFractionITSTPCmatchEtaVsPt[i] = new TEfficiency(Form("mFractionITSTPCmatchEtaVsPt_%s", title[i].c_str()), Form("Fraction of ITSTPC matched tracks, wrt %s tracks, Eta vs Pt; #it{p}_{T} [GeV#it{c}]; Eta; Eff", title[i].c_str()), 100, 0.f, 20.f, 100, -2.f, 2.f);
+    if (mUseTrkPID) { // Vs Tracking PID hypothesis
+      for (int j = 0; j < PID::NIDs; ++j) {
+        // Pt
+        mPtNumVsTrkPID[i][j] = static_cast<TH1D*>(mPtNum[i]->Clone(Form("mPtNumVsTrkPID_%s_PID%i", title[i].c_str(), j)));
+        mPtDenVsTrkPID[i][j] = static_cast<TH1D*>(mPtDen[i]->Clone(Form("mPtDenVsTrkPID_%s_PID%i", title[i].c_str(), j)));
+        mFractionITSTPCmatchPtVsTrkPID[i][j] = static_cast<TEfficiency*>(mFractionITSTPCmatch[i]->Clone(Form("mFractionITSTPCmatchPtVsTrkPID_%s_PID%i", title[i].c_str(), j)));
+        // Phi
+        mPhiNumVsTrkPID[i][j] = static_cast<TH1D*>(mPhiNum[i]->Clone(Form("mPhiNumVsTrkPID_%s_PID%i", title[i].c_str(), j)));
+        mPhiDenVsTrkPID[i][j] = static_cast<TH1D*>(mPhiDen[i]->Clone(Form("mPhiDenVsTrkPID_%s_PID%i", title[i].c_str(), j)));
+        mFractionITSTPCmatchPhiVsTrkPID[i][j] = static_cast<TEfficiency*>(mFractionITSTPCmatchPhi[i]->Clone(Form("mFractionITSTPCmatchPhiVsTrkPID_%s_PID%i", title[i].c_str(), j)));
+        // Eta
+        mEtaNumVsTrkPID[i][j] = static_cast<TH1D*>(mEtaNum[i]->Clone(Form("mEtaNumVsTrkPID_%s_PID%i", title[i].c_str(), j)));
+        mEtaDenVsTrkPID[i][j] = static_cast<TH1D*>(mEtaDen[i]->Clone(Form("mEtaDenVsTrkPID_%s_PID%i", title[i].c_str(), j)));
+        mFractionITSTPCmatchEtaVsTrkPID[i][j] = static_cast<TEfficiency*>(mFractionITSTPCmatchEta[i]->Clone(Form("mFractionITSTPCmatchEtaVsTrkPID_%s_PID%i", title[i].c_str(), j)));
```

Comment:
Thanks! Looks fine but Clone goes via disk IO, why not using constructors?

Reply:
True! Will do!